### PR TITLE
FIXED: Before this PR, when a CPSearchField predicate was bound specifying a predicate format using more than one time the value of the search field ($value), only the first occurence was correctly set, others were set to null

### PR DESCRIPTION
**Pull request by didierkorthoudt**
This was due to how the predicate format was used : all ```$value``` were replaced by ```%@```, then used in ```[CPPredicate predicateWithFormat:formatString, aValue]``` (so specifying only once the value).

The fix simply replaces all ```$value``` directly by the value (surrounded by quotes).

### CODE REVIEW

**Comment by didierkorthoudt** [comment]
milestone=1.1

**Comment by cacaodev** [comment]
Is there a manual test where we can see this ? If not, can you add it to an existing one a provide a new one. tx.

**Comment by mrcarlberg** [comment]
What if the "value" is a double quote (")? I guess that work now but will not after this PR?

**Comment by didierkorthoudt** [comment]
Better solution pushed. This one avoids the problem of double quotes.
One can use manual test `CPSearchFieldPredicateBinding` and test with following predicate format : `name beginswith $value OR name endswith $value`

File: AppKit/CPSearchField.j [hunk 1]
```diff
@@ -915,16 +915,25 @@ var CPRecentsAutosaveNameKey            = @"CPRecentsAutosaveNameKey",
...
 
     if (aValue)
-        [_controller setFilterPredicate:[CPPredicate predicateWithFormat:formatString, aValue]];
+    {
+        var valueCount   = [[_predicateFormat componentsSeparatedByString:@"$value"] count] - 1,
+            formatString = _predicateFormat.replace(/\$value/g, "%@"),
+            values       = @[];
+
+        for (var i = 0; i < valueCount; i++)
+            [values addObject:aValue];
```
Comment (mrcarlberg) [inline AppKit/CPSearchField.j:927]: Nice creative solution here 😃 

What about doing it like this?
```
var values = @[],
    formatString = _predicateFormat.replace(/\$value/g, function(x) {[values addObject:aValue]; return "%@";});             
```

Comment (didierkorthoudt) [reply AppKit/CPSearchField.j:927]: Even more creative !.. 😄
I love this one !.. (I should have found this by myself 😉)
OK, I'll adapt ASAP.

**Comment by mrcarlberg** [comment]
Thanks! Merged!

**Pull request merged**